### PR TITLE
restore old signal handler asap

### PR DESCRIPTION
**Pull request by graingert**
might help with #1310

### CODE REVIEW

File: coverage/control.py [hunk 1]
```diff
@@ -605,10 +605,10 @@ def _atexit(self, event="atexit"):
 
     def _on_sigterm(self, signum_unused, frame_unused):
         """A handler for signal.SIGTERM."""
+        signal.signal(signal.SIGTERM, self._old_sigterm)
         self._atexit("sigterm")
```
Comment (graingert) [inline coverage/control.py:609]: this way if someone is furiously mashing ctrl+c they don't start multiple `self._atexit` calls

Comment (graingert) [reply coverage/control.py:609]: might also be prudent to check if `self._atexit.__code__` is in the `frame_unused` currently

**Comment by graingert** [comment]
This seems to help a lot! However I still got a flake here https://github.com/graingert/distributed/runs/4967453793?check_suite_focus=true#step:13:1240

**Comment by nedbat** [comment]
Thanks, but this doesn't seem to fix #1310.  Detailed reproduction steps are at https://github.com/nedbat/coveragepy/issues/1310#issuecomment-1024178941

**Comment by nedbat** [comment]
I've added a setting to opt-in to SIGTERM handling.

**Pull request closed without merging**